File: LeagueItemExplorerJAR/LeagueItemExplorerJAR/src/main/java/com/app/LeagueItemExplorerJAR/models/ChampionData.java
package com.app.LeagueItemExplorerJAR.models;

import lombok.Data;

@Data
public class ChampionData {
    private String version;
    private String id;
    private String key;
    private String name;
    private String title;
    private String blurb;
    private ChampionBase info;
    private ChampionImage image;
    private String[] tags;
    private String partype;
    private ChampionStats stats;
}

File: LeagueItemExplorerFront/src/app/models/champion-data.ts
import { ChampionBase } from "./champion-base";
import { ChampionImage } from "./champion-image";
import { ChampionStats } from "./champion-stats";

export interface ChampionData {
    version: string;
    id: string;
    key: string;
    name: string;
    title: string;
    blurb: string;
    info: ChampionBase;
    image: ChampionImage;
    tags: string[];
    partype: string;
    stats: ChampionStats;
}